charts/actions-runner-controller/templates/webhook_configs.yaml

```diff
@@ -1,4 +1,8 @@
-
+{{/*
+We will use a self managed CA if one is not provided by cert-manager
+*/}}
+{{- $ca := genCA "actions-runner-ca" 3650 }}
+{{- $cert := genSignedCert (include "actions-runner-controller.webhookDomain" .) nil (list (printf "%s.svc" (include "actions-runner-controller.webhookDomain" .)) (printf "%s.svc.cluster.local" (include "actions-runner-controller.webhookDomain" .))) 3650 $ca }}
```

Comment:
As we're currently using `%s.svc.cluster.local` style host for admission webhooks as you can see in the below snippet:

```
    service:
      name: {{ include "actions-runner-controller.webhookServiceName" . }}
      namespace: {{ .Release.Namespace }}
```

Probably we can just omit the parts related to `webhookDomain`? (If there's any use for it, please correct me!

Reply:
Sure I can clean this part up

Reply:
I updated this to remove the `webhookDomain` bit but we still need to add the name to the SANs due to the way the ARC uses the web hook.

With just the common name I would get errors like this in the ARC log:

```
2022-09-20T17:03:19Z	ERROR	Reconciler error	{"controller": "runnerdeployment-controller", "controllerGroup": "actions.summerwind.dev", "controllerKind": "RunnerDeployment", "runnerDeployment": {"name":"fm-runners","namespace":"actions-runners-system"}, "namespace": "actions-runners-system", "name": "fm-runners", "reconcileID": "a49043d1-5389-4dae-ac08-3f0b5a301efd", "error": "Internal error occurred: failed calling webhook \"mutate.runnerreplicaset.actions.summerwind.dev\": failed to call webhook: Post \"https://actions-runner-controller-webhook.actions-runners-system.svc:443/mutate-actions-summerwind-dev-v1alpha1-runnerreplicaset?timeout=10s\": x509: certificate relies on legacy Common Name field, use SANs instead"}
```

Reply:
Thanks a lot for your investigation! That makes sense. Apparently you now updated it to pass `printf "%s.%s.svc" (include "actions-runner-controller.webhookServiceName" .) .Release.Namespace` to both CN(the first arg of genSignedCert) and SANs(the third arg of genSignedCert), right? Looks good.

Reply:
Yes, we need a valid CN as well as valid SAN.